src/test/javascript/spec/component/team/teams-import-dialog.component.spec.ts

```diff
@@ -0,0 +1,735 @@
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { DebugElement } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick, waitForAsync } from '@angular/core/testing';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { User } from 'app/core/user/user.model';
+import { Exercise } from 'app/entities/exercise.model';
+import { Team, TeamImportStrategyType } from 'app/entities/team.model';
+import { ArtemisTeamModule } from 'app/exercises/shared/team/team.module';
+import { TeamService } from 'app/exercises/shared/team/team.service';
+import { TeamsImportDialogComponent } from 'app/exercises/shared/team/teams-import-dialog/teams-import-dialog.component';
+import * as chai from 'chai';
+import { flatMap } from 'lodash';
+import { JhiAlertService, JhiEventManager } from 'ng-jhipster';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { of, throwError } from 'rxjs';
+import { SinonSpy, SinonStub, spy, stub } from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import {
+    mockExercise,
+    mockSourceExercise,
+    mockSourceTeams,
+    mockSourceTeamStudents,
+    mockTeam,
+    mockTeams,
+    MockTeamService,
+    mockTeamStudents,
+} from '../../helpers/mocks/service/mock-team.service';
+import { ArtemisTestModule } from '../../test.module';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+export const config = {
+    imports: [ArtemisTestModule, ArtemisTeamModule],
+    declarations: [],
+    providers: [
+        JhiEventManager,
+        { provide: TeamService, useClass: MockTeamService },
```

Comment:
Creating a Mock yourself is nice but the big problem is that you always need to adapt your mock if you for example add a new method to the service. Why not use something like https://www.npmjs.com/package/ng-mocks?